repository_service_tuf/cli/admin/metadata.py

```diff
@@ -0,0 +1,605 @@
+    root_table = table.Table()
+    root_table.add_column("Root", justify="left", vertical="middle")
+    root_table.add_column("KEYS", justify="center", vertical="middle")
+    number_of_keys = len(root_info.keys)
+
+    root_keys_table = _create_keys_table(root_info.keys, root_info, True)
+
+    root_table.add_row(
+        (
+            f"\nNumber of Keys: [yellow]{number_of_keys}[/]"
+            f"\nThreshold: [yellow]{root_info.threshold}[/]"
+            f"\nRoot Expiration: [yellow]{root_info.expiration_str}[/]"
+        ),
+        root_keys_table,
+    )
+
+    console.print("\n", root_table)
+    console.print("\n")
+
+
+def _get_key(role: str) -> RSTUFKey:
+    key_type = prompt.Prompt.ask(
+        f"\nChoose [cyan]{role}[/] key type",
+        choices=KeyType.get_all_members(),
+        default=KeyType.KEY_TYPE_ED25519.value,
+    )
+    filepath = prompt.Prompt.ask(
+        f"Enter the [cyan]{role}[/]`s private key [green]path[/]"
+    )
+    colored_role = click.style(role, fg="cyan")
+    colored_pass = click.style("password", fg="green")
+    password = click.prompt(
+        f"Enter the {colored_role}`s private key {colored_pass}",
+        hide_input=True,
+    )
+
+    return load_key(filepath, key_type, password, "")
+
+
+def _is_valid_current_key(
+    keyid: str, current_root: RootInfo, already_loaded_keyids: List[str]
+) -> bool:
+    """Verify that key with `keyid` have been used to sign the current root"""
+    if keyid in already_loaded_keyids:
+        console.print(
+            ":cross_mark: [red]Failed[/]: You already loaded this key",
+            width=100,
+        )
+        return False
+
+    if not current_root.is_keyid_used(keyid):
+        console.print(
+            (
+                ":cross_mark: [red]Failed[/]: This key has not been used "
+                "to sign current root metadata",
+            ),
+            width=100,
+        )
+        return False
+
+    return True
+
+
+def _current_root_keys_validation(current_root: RootInfo):
+    """
+    Authorize user by loading current root threshold number of root keys
+    used for signing the current root metadata.
+    """
+    console.print(markdown.Markdown(AUTHORIZATION), width=100)
+    threshold = current_root.threshold
+    console.print(f"You will need to load {threshold} key(s).")
+    loaded: List[str] = []
+    key_count = 0
+    while key_count < current_root.threshold:
+        console.print(
+            f"You will enter information for key {key_count} of {threshold}"
+        )
+        root_key: RSTUFKey = _get_key(Root.type)
+        if root_key.error:
+            console.print(root_key.error)
+            try_again = prompt.Confirm.ask("Try again?")
+            if try_again:
+                continue
+            else:
+                raise click.ClickException(
+                    f"Failed authorization. Required: {threshold}, loaded "
+                    f"{len(loaded)} keys"
+                )
```

Comment:
As I said before, the user shouldn't be given a choice, where one of the options leads to a failure.

Maybe this is only a wording problem. You could tell them to either try again, or exit, indicating the consequences, i.e. no metadata change or upload. Then, when you do exit, you don't have to say that something failed.

But as I also said before, it seems inconsistent to give the option to exit in a few places but require `ctrl+c` in others.

Reply:
You have a point. Will fix that.